### ISSUE TITLE
Failed to create sender when repair port is not specified

### ISSUE THREAD

**Issue by dshil** [roc-project/roc#220]
How to reproduce:

```
scons -Q --build-3rdparty=openfec,cpputest
roc-send -s rtp:127.0.0.1:10001 -i foo.wav
roc-recv -s rtp:127.0.0.1:10001
```

The reason is that sender is not fully initialized in constructor because of:

```cpp

#ifdef ROC_TARGET_OPENFEC
    if (config.fec_encoder.scheme != packet::FEC_None) {
        if (!repair_port_) {
            return;
        }
```

As a result `sender.valid()` will return false.

**Comment by gavv** [comment]
We also need pipeline-level tests for this case. They didn't catch this issue because currently `sender_receiver` tests always create a receiver with the full set of ports (2x rs, 2x ldpc, 1x rtp). We need to add three additional tests that will create the receiver with only 2x rs, 2x ldpc, or 1x rtp port.

**Comment by gavv** [comment]
Umm, this is not a bug. The sender fails because the `--fec` option is omitted and its default value is `rs`, so the sender expects a repair port but its missing.

So this works:

```
roc-send -s rtp:127.0.0.1:10001 --fec=none -i foo.wav
```

The only problem is that there is no meaningful error message. It will be addressed by #221.
